src/components/CreateEncounter/PatientPrescription.js

```diff
@@ -136,11 +186,24 @@ export const PatientPrescription = ({
     }
   };
 
+  const onLabsIconClick = (name) => {
+    if (!checkLabsAvailability(name)) {
+      onLabsAddClick(name);
+    } else {
+      onLabsRemoveClick(name);
+    }
+  };
+
   const removeMed = (i) => {
     prescriptionList.splice(i, 1);
     setPrescriptionList([...prescriptionList]);
   };
 
+  const removeLab = (i) => {
+    labsList.splice(i, 1);
```

Comment:
create a copy of the labsList and work with it, do not mutate inside of the function, use slice instead https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice